docs/src/misc.rst

```diff
@@ -579,9 +579,9 @@ API
 
     .. versionadded:: 1.25.0
 
-.. c:function:: int uv_gettimeofday(uv_timeval_t* tv)
+.. c:function:: int64_t uv_gettimeofday()
 
-    Cross-platform implementation of :man:`gettimeofday(2)`. The timezone
-    argument to `gettimeofday()` is not supported, as it is considered obsolete.
+    Cross-platform implementation of micro-seconds since epoch. Similar to
+    :man:`gettimeofday(2)`. Negative returned values are error codes.
```

Comment:
So, on many systems it's technically possible to set the system clock to a date before January 1, 1970; that's why `tv_sec` is signed, to support such dates.

I don't know if it's actually useful to support that but in that case libuv should detect such dates and return an error, not just a semi-random value < 0.

I see that you have an `assert(ret > 0)` now. I guess that's okay?

Reply:
IIUC supporting `tv_sec < 0` is mutually exclusive with Y2K38...
I think we should pick a side (either cast to unsigned or bail).

Reply:
This function overlaps too much with `uv_gettimeofday()`, IMO. I'd rather remove it.

> IIUC supporting tv_sec < 0 is mutually exclusive with Y2K38...

Not sure I follow. Not if `tv_sec` is 64 bits.

Reply:
I'd rather remove this function to. Or, at the very least, open a separate PR for it.

I'd like to get the `uv_timeval64_t` changes landed before the next release.

Reply:
I'll spin this off to a new PR.